src/effects/native/reverbeffect.cpp

```diff
@@ -74,48 +99,35 @@ void ReverbEffect::processChannel(const ChannelHandle& handle,
     Q_UNUSED(handle);
     Q_UNUSED(enableState);
     Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
-    CSAMPLE bandwidth = m_pBandWidthParameter->value();
-    CSAMPLE damping = m_pDampingParameter->value();
 
-    // Flip value around.  Assumes max allowable is 1.0.
-    damping = 1.0 - damping;
+    const auto bandwidth = m_pBandWidthParameter->value();
+    const auto decay = m_pDecayParameter->value();
+    const auto damping = m_pDampingParameter->value();
+    const auto blend = m_pBlendParameter->value();
 
-    bool params_changed = (damping != pState->prev_damping ||
-                           bandwidth != pState->prev_bandwidth);
+    // Set the sample rate (TODO XXX: Maybe we want to call .init() after this?)
+    pState->reverb.setSampleRate(sampleRate);
 
-    pState->reverb.setBandwidth(bandwidth);
-    pState->reverb.setDecay(damping);
+    // Process the buffer using the new current parameter values
+    processBuffer(pState->reverb, pInput, pOutput, numSamples, bandwidth, decay, damping, blend);
+}
 
-    for (uint i = 0; i + 1 < numSamples; i += 2) {
-        CSAMPLE mono_sample = (pInput[i] + pInput[i + 1]) / 2;
+void ReverbEffect::processBuffer(MixxxPlateX2& effect, const CSAMPLE* in, CSAMPLE* out, const unsigned int numSamples,
```

Comment:
As a project we've decided to stick to the style guide so that the code is consistent and all parts of the code are readable for all of our developers.  While there are occasionally cases where the style guide can feel restricting or arbitrary, the benefits of having a consistent codebase, especially over very long development periods, makes it worth it to us.  While some old crufty parts of the code may not be conforming, we strongly prefer that new code follow the guide (with our slight modifications (https://www.mixxx.org/wiki/doku.php/coding_guidelines))